### PR TITLE
Add autossh for zuul executor log streaming.

### PR DESCRIPTION
**Pull request by kravvcuCL**

### CODE REVIEW

File: modules/opencontrail_ci/manifests/zuul_executor_autossh.pp [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    owner  => 'root',
+    group  => 'root',
+    mode   => '0644',
+    content => template('opencontrail_ci/autossh-zuul.service.erb'),
+  }
+
+  service { 'autossh-zuul':
+    ensure    => running,
+    enable    => true,
+    subscribe => File['/etc/systemd/system/autossh-zuul.service'],
```
Comment (diabelko) [inline modules/opencontrail_ci/manifests/zuul_executor_autossh.pp]: Subscribe, require, notify, before

File: modules/opencontrail_ci/templates/autossh-zuul.service.erb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+[Unit]
+Description=AutoSSH-based reverse SSH proxy for zuul finger connection
+After=network.target
+
+[Service]
+User=zuul
+Environment="AUTOSSH_GATETIME=0"
+ExecStart=/usr/bin/autossh -i <%= @ssh_key_path %> -M <%= @monitoring_port %> -o "UserKnownHostsFile /dev/null" -o "StrictHostKeyChecking no" -o "ServerAliveInterval 30" -o "ServerAliveCountMax 3" -NR <%= @finger_port %>:localhost:<%= @finger_port %> <%= @ssh_user %>@<%= @zuul_host %>
```
Comment (diabelko) [inline modules/opencontrail_ci/templates/autossh-zuul.service.erb:8]: Does autossh come with the system? If not, we need to install it.

**Review by diabelko — CHANGES_REQUESTED** [review]

File: modules/opencontrail_ci/manifests/zuul_executor_autossh.pp (hunk 1 above)
Comment (kravvcuCL) [reply modules/opencontrail_ci/manifests/zuul_executor_autossh.pp]: Done.

File: modules/opencontrail_ci/templates/autossh-zuul.service.erb (hunk 2 above)
Comment (kravvcuCL) [reply modules/opencontrail_ci/templates/autossh-zuul.service.erb:8]: Done.

**Review by diabelko — APPROVED** [review]

**Review by wurbanski — APPROVED** [review]

**Pull request merged**